lib/Dialect/Torch/Transforms/DecomposeComplexOps.cpp

```diff
@@ -741,6 +797,131 @@ class DecomposeAtenMeanOp : public OpRewritePattern<AtenMeanOp> {
 };
 } // namespace
 
+// productDimSize = product(size(dim) for dim in dims)
+// aten.mean(x, dims) = aten.sum(x, dims) / productDimSize.
+namespace {
+class DecomposeAtenMeanDimOp : public OpRewritePattern<AtenMeanDimOp> {
+public:
+  using OpRewritePattern::OpRewritePattern;
+  LogicalResult matchAndRewrite(AtenMeanDimOp op,
+                                PatternRewriter &rewriter) const override {
+    Location loc = op.getLoc();
+    Value input = op.self();
+    Value dimList = op.dim();
+    Value keepDim = op.keepdim();
+    Type outputType = op.getType();
+
+    BaseTensorType inputType = input.getType().cast<BaseTensorType>();
+    if (!inputType.hasDtype() || !inputType.getDtype().isa<mlir::FloatType>()) {
```

Comment:
I think you should also check that `op.dtype()` is float. I tried passing a type different from float, and PyTorch gives casting errors.